cpp/include/kvikio/error.hpp

```diff
@@ -75,8 +75,33 @@ struct CUfileException : public std::runtime_error {
                              cufileop_status_error(error.err)};                  \
     }                                                                            \
   } while (0)
-#define CUFILE_TRY_1(_call) CUFILE_TRY_2(_call, CUfileException)
+#define CUFILE_TRY_1(_call) CUFILE_TRY_2(_call, kvikio::CUfileException)
 #endif
 #endif
 
+#ifndef CUFILE_CHECK_STREAM_IO
+#define CUFILE_CHECK_STREAM_IO(...)                                  \
```

Comment:
Add a docstring/comment? While we're here. Does this need to be a macro or would a `static inline` function be fine? Then you wouldn't have worries about variable capture/etc...:

```
static inline void check_stream_io(ssize_t nbytes_written) {
   ...
}
```
Or do you want to be able to report line numbers easily?

Reply:
I like the line numbers. In a follow-up, I will look at a more user friendly API for the deferred arguments and errors.